### PR TITLE
added JWT support

### PR DESCRIPTION
**Pull request by apanyutin**

### CODE REVIEW

File: src/main/java/com/example/config/OpenApiConfig.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import io.swagger.v3.oas.models.Components;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.security.SecurityRequirement;
+import io.swagger.v3.oas.models.security.SecurityScheme;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class OpenApiConfig {
+
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/config/OpenApiConfig.java:15]:
```suggestion
```

File: src/main/java/com/example/config/OpenApiConfig.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @Bean
+    public OpenAPI customOpenApi() {
+        return new OpenAPI()
+                .components(new Components().addSecuritySchemes("BearerAuth",
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/config/OpenApiConfig.java]: create constant for strings

File: src/main/java/com/example/security/JwtAuthFilter.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthFilter extends OncePerRequestFilter {
+    private static final String BEARER_BEGINNING = "Bearer ";
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/security/JwtAuthFilter.java]:
```suggestion
    private static final String TOKEN_HEADER = "Bearer ";
```

File: src/main/java/com/example/dto/user/UserLoginResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.example.dto.user;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/dto/user/UserLoginResponseDto.java]:
```suggestion
@Getter
@Setter
```

check if you need these annotations

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
good job! Let's improve it

**Review by sarakhmen — APPROVED** [review]

**Pull request merged**